src/hssm/data_validator.py

```diff
@@ -0,0 +1,205 @@
+            warnings.warn(
+                (
+                    f"You set choices to be {self.choices}, but {missing_responses} "
+                    "are missing from your dataset."
+                ),
+                UserWarning,
+                stacklevel=2,
+            )
+
+    def _handle_missing_data_and_deadline(self):
+        """Handle missing data and deadline."""
+        if not self.missing_data and not self.deadline:
+            # In the case where missing_data is set to False, we need to drop the
+            # cases where rt = na_value
+            if pd.isna(self.missing_data_value):
+                na_dropped = self.data.dropna(subset=["rt"])
+            else:
+                na_dropped = self.data.loc[
+                    self.data["rt"] != self.missing_data_value, :
+                ]
+
+            if len(na_dropped) != len(self.data):
+                warnings.warn(
+                    "`missing_data` is set to False, "
+                    + "but you have missing data in your dataset. "
+                    + "Missing data will be dropped.",
+                    stacklevel=2,
+                )
+            self.data = na_dropped
+
+        elif self.missing_data and not self.deadline:
+            # In the case where missing_data is set to True, we need to replace the
+            # missing data with a specified na_value
+
+            # Create a shallow copy to avoid modifying the original dataframe
+            if pd.isna(self.missing_data_value):
+                self.data["rt"] = self.data["rt"].fillna(-999.0)
+            else:
+                self.data["rt"] = self.data["rt"].replace(
+                    self.missing_data_value, -999.0
+                )
+
+        else:  # deadline = True
+            if self.deadline_name not in self.data.columns:
+                raise ValueError(
+                    "You have specified that your data has deadline, but "
+                    + f"`{self.deadline_name}` is not found in your dataset."
+                )
+            else:
+                self.data.loc[:, "rt"] = np.where(
+                    self.data["rt"] < self.data[self.deadline_name],
+                    self.data["rt"],
+                    -999.0,
+                )
+
+    def _update_extra_fields(self, new_data: pd.DataFrame | None = None):
+        """Update the extra fields data in self.model_distribution.
+
+        Parameters
+        ----------
+        new_data
+            A DataFrame containing new data for update.
+        """
+        if not new_data:
+            new_data = self.data
+
+        # The attribute 'model_distribution' is not defined in DataValidator itself,
+        # but is expected to exist in subclasses (e.g., HSSM).
+        # The 'type: ignore[attr-defined]' comment tells mypy to ignore the missing
+        # attribute error here and avoid moving this method to the HSSM class.
+        self.model_distribution.extra_fields = [  # type: ignore[attr-defined]
```

Comment:
I think I resolved cases like this with `if hasattr()` before. But not sure what the recommended pattern is.